### PR TITLE
Add pymodbus recipe

### PR DESCRIPTION
**Pull request by saibhargav1508**
Checklist
- 

- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

I am willing to be listed in the recipe as maintainer

### CODE REVIEW

**Comment by saibhargav1508** [comment]
I am willing to be listed in the recipe as maintainer

File: recipes/pymodbus/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - pyserial >=3.4
+    - python
+    - six >=1.11.0
+
+test:
+  imports:
+    - pymodbus
+    - pymodbus.client
+  requires:
+    - pip
```
Comment (marcelotrevisani) [inline recipes/pymodbus/meta.yaml:34]:
```suggestion
    - pip
  commands:
    - pymodbus.console --help
    - pip check
```

Comment (saibhargav1508) [reply recipes/pymodbus/meta.yaml:34]: Could you please tell why these additional commands are required?

**Comment by saibhargav1508** [comment]
Need some help.Don't seem to figure out what is the issue with the test cases

**Comment by marcelotrevisani** [comment]
> Need some help.Don't seem to figure out what is the issue with the test cases

It seems ``click`` is a dependency for this project, so we will need to add it
The maintainers didn't add it as a explicit dependency
lets fix it :)

File: recipes/pymodbus/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - pyserial >=3.4
+    - python
+    - six >=1.11.0
```
Comment (marcelotrevisani) [inline recipes/pymodbus/meta.yaml:27]:
```suggestion
    - six >=1.11.0
    - click
```

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**